ethcore/light/src/provider.rs

```diff
@@ -143,67 +143,87 @@ pub trait Provider: Send + Sync {
+		trace!(target: "pip_provider", " chain_info {:?}", chain_info);
+		chain_info
 	}
 
 	fn reorg_depth(&self, a: &H256, b: &H256) -> Option<u64> {
-		self.tree_route(a, b).map(|route| route.index as u64)
+		let reorg = self.tree_route(a, b).map(|route| route.index as u64);
+		trace!(target: "pip_provider", " reorg_depth {:?}", reorg);
+		reorg
 	}
 
 	fn earliest_state(&self) -> Option<u64> {
-		Some(self.pruning_info().earliest_state)
+		let earliest_state = self.pruning_info().earliest_state;
+		trace!(target: "pip_provider", "earliest_state {:?}", earliest_state);
+		Some(earliest_state)
 	}
 
 	fn block_header(&self, id: BlockId) -> Option<encoded::Header> {
-		ClientBlockInfo::block_header(self, id)
+		let block_header = ClientBlockInfo::block_header(self, id);
+		trace!(target: "pip_provider", "block_header {:?}", block_header);
+		block_header
 	}
 
 	fn transaction_index(&self, req: request::CompleteTransactionIndexRequest)
 		-> Option<request::TransactionIndexResponse>
 	{
 		use ethcore::ids::TransactionId;
 
-		self.transaction_receipt(TransactionId::Hash(req.hash)).map(|receipt| request::TransactionIndexResponse {
+		let transaction_receipt = self.transaction_receipt(TransactionId::Hash(req.hash)).map(|receipt| request::TransactionIndexResponse {
 			num: receipt.block_number,
 			hash: receipt.block_hash,
 			index: receipt.transaction_index as u64,
-		})
+		});
+		trace!(target: "pip_provider", "transaction_receipt: {:?}", transaction_receipt);
+		transaction_receipt
 	}
 
 	fn block_body(&self, req: request::CompleteBodyRequest) -> Option<request::BodyResponse> {
-		BlockChainClient::block_body(self, BlockId::Hash(req.hash))
-			.map(|body| ::request::BodyResponse { body })
+		let block_body = BlockChainClient::block_body(self, BlockId::Hash(req.hash))
+			.map(|body| ::request::BodyResponse { body });
+		trace!(target: "pip_provider", "block_body: {:?}", block_body);
+		block_body
 	}
 
 	fn block_receipts(&self, req: request::CompleteReceiptsRequest) -> Option<request::ReceiptsResponse> {
-		BlockChainClient::encoded_block_receipts(self, &req.hash)
-			.map(|x| ::request::ReceiptsResponse { receipts: ::rlp::decode_list(&x) })
+		let block_receipt = BlockChainClient::encoded_block_receipts(self, &req.hash)
+			.map(|x| ::request::ReceiptsResponse { receipts: ::rlp::decode_list(&x) });
+		trace!(target: "pip_provider", "block_receipt: {:?}", block_receipt);
+		block_receipt
 	}
 
 	fn account_proof(&self, req: request::CompleteAccountRequest) -> Option<request::AccountResponse> {
-		self.prove_account(req.address_hash, BlockId::Hash(req.block_hash)).map(|(proof, acc)| {
+		let account_proof = self.prove_account(req.address_hash, BlockId::Hash(req.block_hash)).map(|(proof, acc)| {
 			::request::AccountResponse {
 				proof,
 				nonce: acc.nonce,
 				balance: acc.balance,
 				code_hash: acc.code_hash,
 				storage_root: acc.storage_root,
 			}
-		})
+		});
+		trace!(target: "pip_provider", "account_proof: {:?}", account_proof);
```

Comment:
Well, I don't think is necessary to print out all the raw bytes here so I will change these to be more "human friendly"